index.ts

```diff
@@ -75,199 +65,254 @@ class ServerlessCustomDomain {
-        const restApiId = await this.getRestApiId();
-        const currentBasePath = await this.getBasePathMapping(restApiId);
-        // if basepath that matches restApiId exists, update; else, create
-        if (!currentBasePath) {
-            await this.createBasePathMapping(restApiId);
-        } else {
-            await this.updateBasePathMapping(currentBasePath);
-        }
-        const domainInfo = await this.getDomainInfo();
-        await this.printDomainSummary(domainInfo);
+    public async setupBasePathMappings(): Promise<void> {
+        await Promise.all(this.domains.map(async (domain) => {
+            try {
+                domain.apiId = await this.getApiId(domain);
+
+                domain.apiMapping = await this.getBasePathMapping(domain);
+
+                if (!domain.apiMapping) {
+                    await this.createBasePathMapping(domain);
+                } else {
+                    await this.updateBasePathMapping(domain);
+                }
+
+                await this.getDomainInfo();
+
+            } catch (err) {
+                this.logIfDebug(err, domain.givenDomainName);
+                throw new Error(`Error: Unable to setup base domain mappings for ${domain.givenDomainName}`);
+            }
+        })).then(() => {
+            // Print summary upon completion
+            this.domains.forEach((domain) => {
+                this.printDomainSummary(domain);
+            });
+        });
     }
 
     /**
      * Lifecycle function to delete basepath mapping
      * Wraps deletion of basepath mapping
      */
-    public async removeBasePathMapping(): Promise<void> {
-        await this.deleteBasePathMapping();
+    public async removeBasePathMappings(): Promise<void> {
+        await Promise.all(this.domains.map(async (domain) => {
+            try {
+                domain.apiId = await this.getApiId(domain);
+
+                // Unable to find the corresponding API, manual clean up will be required
+                if (!domain.apiId) {
+                    this.serverless.cli.log(`Unable to find corresponding API for ${domain.givenDomainName},
+                        API Mappings may need to be manually removed.`, "Serverless Domain Manager");
+                } else {
+                    domain.apiMapping = await this.getBasePathMapping(domain);
+                    await this.deleteBasePathMapping(domain);
+                }
+            } catch (err) {
+                if (err.message.indexOf("Failed to find CloudFormation") > -1) {
+                    this.serverless.cli.log(`Unable to find Cloudformation Stack for ${domain.givenDomainName},
+                        API Mappings may need to be manually removed.`, "Serverless Domain Manager");
+                } else {
+                    this.logIfDebug(err, domain.givenDomainName);
+                    throw new Error(`Error: Unable to remove base bath mappings for domain ${domain.givenDomainName}`);
```

Comment:
This is actually changing some of the old behavior. This used to log an error without throwing an exception. I ran into this by accident by running `delete_domain` and `remove` out of order. For example:

```
sls create_domain
sls deploy
sls delete_domain
sls remove
```

The `sls remove` command above started throwing errors and I wasn't able to remove the API until running `sls create_domain` and `sls_deploy` again. 

I see the desire to be more strict with the error handling so can either change this back to a log statement or add a note to the readme.

Reply:
Hmm...ya...I'm looking at that and can't recall a good reason why. I think I was just following the same approach used in other try/catch blocks and may have been a copy/paste error.

I removed the `throw` and added a log.